### PR TITLE
Add support for tailoring of external datastreams

### PR DESCRIPTION
**Pull request by jan-cerny**
This pull request allows user to evaluate a tailoring within datastream that customizes benchmarks located in other datastream in on other datastream collection in an other separate file.

### CODE REVIEW

File: src/XCCDF/xccdf_session.c [hunk 1]
```diff
@@ -524,13 +524,33 @@ static int _acquire_xccdf_checklist_from_tailoring(struct xccdf_session* session
...
+	struct oscap_source *xccdf_source = NULL;
+	if (oscap_str_startswith(benchmark_ref, "file:")) {
+		char* sep = strchr(benchmark_ref, '#');
+		const char *filename = benchmark_ref + strlen("file:");
+		const char *component_ref = sep + 1;
+		*sep = '\0';
+
+		struct oscap_source *external_file = oscap_source_new_from_file(filename);
+		if (oscap_source_get_scap_type(external_file) == OSCAP_DOCUMENT_SDS) {
+			struct ds_sds_session *ds_sds_session_tmp = ds_sds_session_new_from_source(external_file);
```
Comment (mpreisler) [inline src/XCCDF/xccdf_session.c]: do you really need an extra variable here? why not just free ds_sds_session and reallocate it?

Comment (jan-cerny) [inline src/XCCDF/xccdf_session.c]: yeah, great, thanks, that would be better and more readable

File: src/XCCDF/xccdf_session.c [hunk 2]
```diff
@@ -524,13 +524,33 @@ static int _acquire_xccdf_checklist_from_tailoring(struct xccdf_session* session
 		xmlFree(tailoring_xmlDoc);
 		return 1;
 	}
-	char *benchmark_id = strchr(benchmark_ref, '#') + 1;
 
-	struct oscap_source *xccdf_source = ds_sds_session_select_checklist(ds_sds_session,
-			NULL, benchmark_id, NULL);
+	struct oscap_source *xccdf_source = NULL;
+	if (oscap_str_startswith(benchmark_ref, "file:")) {
+		char* sep = strchr(benchmark_ref, '#');
```
Comment (jan-cerny) [inline src/XCCDF/xccdf_session.c]: Ok, I fix this below.

**Comment by jan-cerny** [comment]
I have improved this. I have added checks for NULL and I have removed a temporary variable.

**Comment by ybznek** [comment]
I think we should have test for this. Or do you think that we can use NIST testuite to be test for it?

File: src/XCCDF/xccdf_session.c [hunk 3]
```diff
@@ -524,13 +524,43 @@ static int _acquire_xccdf_checklist_from_tailoring(struct xccdf_session* session
...
+		const char *filename = benchmark_ref + strlen("file:");
+		const char *component_ref = NULL;
+		if (sep != NULL) {
+			component_ref = sep + 1;
+			*sep = '\0';
+		}
+
+		struct oscap_source *external_file = oscap_source_new_from_file(filename);
+		if (oscap_source_get_scap_type(external_file) == OSCAP_DOCUMENT_SDS) {
```
Comment (ybznek) [inline src/XCCDF/xccdf_session.c]: It is good to check, could you please add this condition to sds.c where can be found similar code?

Comment (jan-cerny) [inline src/XCCDF/xccdf_session.c]: @ybznek yes, I can, but let's do that in other PR

**Comment by ybznek** [comment]
LGTM, thank you!

**Pull request merged**